core/node/api_server/src/tx_sender/mod.rs

```diff
@@ -1036,4 +1039,19 @@ impl TxSender {
         }
         Ok(())
     }
+
+    pub(crate) async fn get_default_eth_call_gas(
+        &self,
+        block_args: BlockArgs,
+    ) -> anyhow::Result<u64> {
+        let mut connection = self.acquire_replica_connection().await?;
+
+        let protocol_version = block_args
+            .resolve_block_info(&mut connection)
+            .await
+            .context("failed to resolve block info")?
+            .protocol_version;
```

Comment:
Nit / future work: It may make sense to create a dedicated method for resolving the protocol version since `resolve_block_info` loads a lot of unnecessary stuff from the DB.

Reply:
I would leave it for future work